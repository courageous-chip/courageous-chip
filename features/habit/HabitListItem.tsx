import { gql } from "@apollo/client";
import React, { FC } from "react";
import { Text, VirtualizedListProps } from "react-native";

import { HabitListItemFields } from "./__generated__/HabitListItemFields";

export const HABIT_LIST_ITEM_FIELDS_FRAGMENT = gql`
  fragment HabitListItemFields on Habit {
    id
    name
  }
`;

export const keyExtractor: VirtualizedListProps<
  HabitListItemFields
>["keyExtractor"] = function ({ id }) {
  return id;
};

export const renderItem: VirtualizedListProps<
  HabitListItemFields
>["renderItem"] = function ({ item: { name } }) {
  return <HabitListItem name={name} />;
};

type Props = Pick<HabitListItemFields, "name">;

export const HabitListItem: FC<Props> = function ({ name }) {
  return <Text testID="habitItemTestID">{name}</Text>;
};
