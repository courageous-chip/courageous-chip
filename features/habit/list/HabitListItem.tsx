import { gql } from "@apollo/client";
import React, { FC } from "react";
import { FlatListProps, StyleSheet, Text, View } from "react-native";

import { HabitListItemFields } from "./__generated__/HabitListItemFields";

export const HABIT_LIST_ITEM_FIELDS_FRAGMENT = gql`
  fragment HabitListItemFields on Habit {
    id
    name
  }
`;

export const ItemSeparatorComponent: FlatListProps<
  HabitListItemFields
>["ItemSeparatorComponent"] = function () {
  return <View style={styles.itemSeparator} />;
};

export const keyExtractor: FlatListProps<
  HabitListItemFields
>["keyExtractor"] = function ({ id }) {
  return id;
};

export const renderItem: FlatListProps<
  HabitListItemFields
>["renderItem"] = function ({ item: { name } }) {
  return <HabitListItem name={name} />;
};

type Props = Pick<HabitListItemFields, "name">;

export const HabitListItem: FC<Props> = function ({ name }) {
  return (
    <View style={styles.itemContainer}>
      <Text style={styles.itemText}>{name}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  itemContainer: { justifyContent: "center", minHeight: 64 },
  itemSeparator: { backgroundColor: "#f7f7f7", height: 1, width: "100%" },
  itemText: { marginHorizontal: 20 },
});
