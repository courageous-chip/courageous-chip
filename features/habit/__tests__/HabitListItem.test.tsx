import { render } from "@testing-library/react-native";
import React from "react";
import renderer from "react-test-renderer";

import { HabitListItem } from "../HabitListItem";

describe("<HabitListItem />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<HabitListItem name="ðŸ¦· Floss Teeth" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

test("habit list item exists", async () => {
  const { getByTestId } = render(<HabitListItem name="Floss" />);

  const val = getByTestId("habitItemTestID").props.name;

  expect(val).toEqual(getByTestId("habitItemTestID").props.name);
});
