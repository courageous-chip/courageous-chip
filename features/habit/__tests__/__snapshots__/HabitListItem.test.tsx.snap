// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitListItem /> renders correctly 1`] = `
<Text
  testID="habitItemTestID"
>
  ðŸ¦· Floss Teeth
</Text>
`;
